src/nvim/eval.c

```diff
@@ -6177,9 +6177,15 @@ bool read_blob(FILE *const fd, blob_T *const blob)
 /// @param[in]  endnl If true, the output will end in a newline (if a list).
 /// @returns an allocated string if `tv` represents a VimL string, list, or
 ///          number; NULL otherwise.
-char *save_tv_as_string(typval_T *tv, ptrdiff_t *const len, bool endnl)
+char *save_tv_as_string(typval_T *tv, ptrdiff_t *const len, bool endnl, bool is_chanterm)
   FUNC_ATTR_MALLOC FUNC_ATTR_NONNULL_ALL
 {
+#ifdef UNIX
+  bool crlf = false;
+#else
+  bool crlf = is_chanterm ? true: false;
+#endif
```

Comment:
> Can we add a crlf paramter to save_tv_as_string which is set to true by f_chansend?

Did something similar^. But I can do crlf as parameter directly as well, if you want.

Reply:
This assumes that no other callers will call `save_tv_as_string`. Seems better to make it explicit?

Reply:
> Seems better to make it explicit?

True. I committed and pushed.

Can't figure out the EOF crash, though. If you had any idea.

Reply:
Oh wait. It passed those tests now! Huh.